cmd/repo.go

```diff
@@ -477,17 +498,20 @@ func (r *RepositoryFile) WriteFile(path string) error {
 }
 
 func (r *RepositoryFile) GetIndices() (RepoIndices, error) {
-
 	indices := make(map[string]*RepoIndex)
+	brokenRepos := make([]indexError, 0)
 	for _, rf := range r.Repositories {
 		var index, err = downloadIndex(rf.URL)
 		if err != nil {
-			return indices, err
+			repoErr := indexError{rf.Name, err}
+			brokenRepos = append(brokenRepos, repoErr)
+		} else {
+			indices[rf.Name] = index
 		}
-		indices[rf.Name] = index
 	}
-	return indices, nil
+	return indices, &indexErrors{brokenRepos}
```

Comment:
I think you should return  `&indexErrors{brokenRepos}` only if `len(brokenRepos) > 0`, otherwise the caller will think there's always an error. Let me know if I am mistaken.

Reply:
I agree, if there are no broken repos, you should return a `nil` error

Reply:
Done